src/Dialect/Krnl/KrnlOps.td

```diff
@@ -595,3 +595,38 @@ def KrnlStoreOp : Op<Krnl_Dialect, "store",
     $value `,` $memref `[` $indices `]` attr-dict `:` type($memref)
   }];
 }
+
+def KrnlMovableOp : Op<Krnl_Dialect, "movable", [ImplicitKrnlTerminator]> {
+  let summary = "Krnl movable operation";
+  let description = [{
+     Encapsulates a list of operations, which should be moved under a newly lowered
+     affine for operation eventually, but cannot presently because the destination
+     affine for operation is not materialized yet.
+  }];
+
+  let arguments = (ins );
+
+  let regions = (region AnyRegion:$region);
+
+  let assemblyFormat = [{
+      $region attr-dict
+  }];
+
+}
+
+def KrnlGetInductionVariableValueOp : Op<Krnl_Dialect, "get_induction_var_value"> {
+  let summary = "Krnl ";
+  let description = [{
+     Krnl operation to convert loop references to corresponding induction
+     variable values. This is useful for accessing optimized loop induction
+     variables, as they are not otherwise accessible during Krnl Dialect.
+  }];
```

Comment:
Can you be a bit more specific on the value returned in a blocked loop? It is not the block id, but the index value corresponding to the block?

Reply:
Ahh, good point. Yes, it's the tile starting indices.